### PR TITLE
Stop using pip on EL7 and Fedora systems

### PR DESCRIPTION
**Pull request by jmontleon**

### CODE REVIEW

**Comment by jmontleon** [comment]
Tested with EL7, Fedora 25, and Fedora 26

File: ansible/roles/openshift_setup/tasks/main.yml [hunk 1]
```diff
@@ -157,6 +157,28 @@
...
+      description: "Ansible Service Broker"
+      baseurl: "https://copr-be.cloud.fedoraproject.org/results/@ansible-service-broker/ansible-service-broker/epel-7-$basearch/"
+      enabled: yes
+      gpgcheck: no
+    when: (ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux') and
+          ansible_distribution_major_version == "7"
+
+  - name: Enable ASB repo for Fedora 25
+    shell: dnf -y copr enable @ansible-service-broker/ansible-service-broker
+    when: ansible_distribution == 'Fedora' and ansible_distribution_major_version == "25"
```
Comment (jwmatthews) [inline ansible/roles/openshift_setup/tasks/main.yml]: I needed to add a 'become: true'

to address below when I invoked 'reset_environment.sh' as my regular user.

fatal: [localhost]: FAILED! => {"changed": true, "cmd": "dnf -y copr enable @ansible-service-broker/ansible-service-broker", "delta": "0:00:00.255520", "end": "2017-08-10 08:56:42.478085", "failed": true, "rc": 1, "start": "2017-08-10 08:56:42.222565", "stderr": "Error: This command has to be run under the root user.", "stderr_lines": ["Error: This command has to be run under the root user."], "stdout": "", "stdout_lines": []}

File: ansible/roles/openshift_setup/tasks/main.yml [hunk 2]
```diff
@@ -157,6 +157,28 @@
...
+
+  - name: Create ASB repo for EL7
+    yum_repository:
+      name: "ansible-service-broker"
+      description: "Ansible Service Broker"
+      baseurl: "https://copr-be.cloud.fedoraproject.org/results/@ansible-service-broker/ansible-service-broker/epel-7-$basearch/"
+      enabled: yes
+      gpgcheck: no
+    when: (ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux') and
+          ansible_distribution_major_version == "7"
```
Comment (jwmatthews) [inline ansible/roles/openshift_setup/tasks/main.yml]: Assume we should add a 'become: true' here as well.

File: ansible/roles/packages/tasks/main.yml [hunk 3]
```diff
@@ -1,4 +1,18 @@
...
+  - name: Create ASB repo for EL7
+    yum_repository:
+      name: "ansible-service-broker"
+      description: "Ansible Service Broker"
+      baseurl: "https://copr-be.cloud.fedoraproject.org/results/@ansible-service-broker/ansible-service-broker/epel-7-$basearch/"
+      enabled: yes
+      gpgcheck: no
+    when: (ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux') and
+          ansible_distribution_major_version == "7"
+
```
Comment (jwmatthews) [inline ansible/roles/packages/tasks/main.yml]: become: true

File: ansible/roles/packages/tasks/main.yml [hunk 4]
```diff
@@ -1,4 +1,18 @@
...
+  - name: Enable ASB repo for Fedora 25
+    shell: dnf -y copr enable @ansible-service-broker/ansible-service-broker
+    when: ansible_distribution == 'Fedora' and ansible_distribution_major_version == "25"
+
```
Comment (jwmatthews) [inline ansible/roles/packages/tasks/main.yml]: become: true

**Comment by jwmatthews** [comment]
Testing on F25.

## Prior to reset_environment.sh

sudo pip list | grep docker
docker (2.3.0)
docker-pycreds (0.2.1)

sudo pip uninstall docker docker-pycreds

$ rpm -qa | grep docker
python2-dockerfile-parse-0.0.5-7.fc25.noarch
docker-1.12.6-6.gitae7d637.fc25.x86_64
docker-common-1.12.6-6.gitae7d637.fc25.x86_64

## After reset_environment.sh

$ rpm -qa | grep docker
python2-dockerfile-parse-0.0.5-7.fc25.noarch
python2-docker-pycreds-0.2.1-2.fc25.noarch
docker-1.12.6-6.gitae7d637.fc25.x86_64
docker-common-1.12.6-6.gitae7d637.fc25.x86_64
python2-docker-2.4.2-1.fc25.noarch

$ sudo pip list | grep docker
docker (2.4.2)
docker-pycreds (0.2.1)




I needed to add a 'become: true' to the enable copr task on F25, otherwise ran through fine.
Verified provision/bind mediawiki and postgres succeeded.

**Comment by jwmatthews** [comment]
Tested on mac and saw success 

Ran a ```pip uninstall docker docker-pycreds``` prior

Verified successful provision/bind workflow

**Comment by jmontleon** [comment]
@jwmatthews I added the come: true statements to those 4 repo creation commands.

**Comment by jwmatthews** [comment]
Tested on ec-2 and saw an error:

TASK [aws_repo_setup : install EPEL] ****************************************************************************************************************************************************
changed: [34.203.86.190]

TASK [aws_repo_setup : Import EPEL GPG key.] ********************************************************************************************************************************************
changed: [34.203.86.190]

TASK [aws_repo_setup : Enable extras repo] **********************************************************************************************************************************************
changed: [34.203.86.190]

TASK [aws_repo_setup : Enable optional repo] ********************************************************************************************************************************************
changed: [34.203.86.190]

TASK [packages : Create ASB repo for EL7] ***********************************************************************************************************************************************
skipping: [34.203.86.190]

TASK [packages : Enable ASB repo for Fedora 25] *****************************************************************************************************************************************
skipping: [34.203.86.190]

TASK [packages : install packages] ******************************************************************************************************************************************************
failed: [34.203.86.190] (item=[u'deltarpm', u'wget', u'git', u'net-tools', u'bind-utils', u'bridge-utils', u'gcc', u'docker', u'vim', u'vim-enhanced', u'golang', u'nodejs', u'ansible', u'python2-docker']) => {"changed": false, "failed": true, "item": ["deltarpm", "wget", "git", "net-tools", "bind-utils", "bridge-utils", "gcc", "docker", "vim", "vim-enhanced", "golang", "nodejs", "ansible", "python2-docker"], "msg": "No package matching 'python2-docker' found available, installed or updated", "rc": 126, "results": ["net-tools-2.0-0.17.20131004git.el7.x86_64 providing net-tools is already installed", "No package matching 'python2-docker' found available, installed or updated"]}

**Comment by jmontleon** [comment]
@jwmatthews I think I fixed the ansible_distribution for RHEL 7. The example I saw had it as 'Red Hat Enterprise Linux', but it looks like it comes back as 'RedHat'.

**Comment by jwmatthews** [comment]
Saw success with ec2

ACK

**Pull request merged**